### ISSUE TITLE
Extraction hangs on specific image

### ISSUE THREAD

**Issue by SilasLaspada** [aaru-dps/Aaru#590]
### Prerequisites

* [X] Are you running the latest version?
* [X] Can you reproduce the problem in the debug version?

### Check and fill as appropriate:
* [X] I was running Aaru under Windows 10 21H1...
* [ ] I was running Aaru under Linux [version here]...
* [ ] I was running Aaru under macOS [version here]...
* [ ] ...in 32-bit
* [X] ...in 64-bit
* [X] ...using an Intel / AMD processor
* [ ] ...using an ARM or Apple Silicon processor
    * **Image format**:  Multiple
    * **Application that created the image**: Multiple
* [ ] I was converting the image...
    * ...**Destination image format**:
* [ ] I was listing the filesystem contents...
* [X] I was extracting the filesystem contents...
    * ...**Filesystem**: ISO9660
    * [X] Analyze command recognizes the filesystem
* [ ] I was creating a metadata sidecar
* [ ] I was checksumming the image
* [ ] I was verifying the image

### Description

When extracting the files from a specific image, Aaru hangs without crashing.

### Exact command line used:

`Aaru fs extract "MB Support CD.aaruf" 0`

### Output of command execution with debug output enabled
```
.....
Written 4706304 bytes of file RTKSMUIInstall64.msi to 0\ISO9660\MB Support CD\Drivers/Audio/Driver/SonicRadar\RTKSMUIInstall64.msi
DEBUG (Aaru Format plugin): Memory snapshot: 527278360 bytes
DEBUG (Aaru Format plugin): Memory snapshot: 575456952 bytes
DEBUG (Aaru Format plugin): Memory snapshot: 575456952 bytes
DEBUG (ISO9660 Plugin): Sector 622236, Cooked, Mode 0/1 / Mode 2 Form 1
Written 553 bytes of file USetup.iss to 0\ISO9660\MB Support CD\Drivers/Audio/Driver\USetup.iss
Created subdirectory at 0\ISO9660\MB Support CD\Drivers/Audio/Driver\Vista
DEBUG (ISO9660 Plugin): Sector 1621, Cooked, Mode 0/1 / Mode 2 Form 1
DEBUG (ISO9660 Plugin): Sector 1622, Cooked, Mode 0/1 / Mode 2 Form 1
DEBUG (ISO9660 Plugin): Sector 1623, Cooked, Mode 0/1 / Mode 2 Form 1
DEBUG (ISO9660 Plugin): Sector 1624, Cooked, Mode 0/1 / Mode 2 Form 1
DEBUG (ISO9660 Plugin): Sector 1625, Cooked, Mode 0/1 / Mode 2 Form 1
DEBUG (ISO9660 Plugin): Sector 1626, Cooked, Mode 0/1 / Mode 2 Form 1
```

### Image

https://archive.org/details/m5a99fx-pro-r2.0-motherboard-support-dvd-amd-990fx-sb950-chipset-rev-611.08-2013

**Comment by SilasLaspada** [comment]
Other images seemingly affected by the same bug include https://archive.org/details/windows-3.0-international-versions-promotional-copy and https://archive.org/details/asus-driver-gpu-tweak-v1231-2014

**Comment by claunia** [comment]
Fixed in a39258524cf5561c75ed824a359b2551a39dee20.
